spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/HandlerMethodDetect.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.listener.adapter;
+
+/**
+ * Auto-detect {@link HandlerAdapter} args and return type.
+ *
+ * @author Wang zhiyang
+ * @since 3.2
+ */
+public interface HandlerMethodDetect {
```

Comment:
I'm not sure in this abstraction and probably its name does not reflect the purpose.
Would you mind elaborating?
Spring AMQP just does that in its `HandlerAdapter` and that's it.

Reply:
I want use it in MessageListenerAdapter, like `FilteringBatchMessageListenerAdapter`.
let @KafkaListener can aware HandlerAdapter args

Reply:
Let's call this contract then like `AsyncRepliesAware`!
Otherwise it is not clear what this supposed to "detect".
The goal of the interface is to give some aspect to the target message handler adapter.